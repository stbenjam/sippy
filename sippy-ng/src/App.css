.App {
  text-align: center;
}

.App-logo {
  height: 10vmin;
}

@media (prefers-reduced-motion: no-preference) {
  .App-logo:hover {
    animation: App-logo-spin infinite 7s linear;
  }
}

.App-link {
  color: #61dafb;
}

@keyframes App-logo-spin {
  from {
    transform: rotate(0deg);
  }
  to {
    transform: rotate(360deg);
  }
}
