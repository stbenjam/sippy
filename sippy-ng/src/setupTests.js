/** @jest-environment setup-polly-jest/jest-environment-node */

import '@testing-library/jest-dom'
import Enzyme from 'enzyme'
import Adapter from '@wojtekmaj/enzyme-adapter-react-17'
import { Polly } from '@pollyjs/core'
import HttpAdapter from '@pollyjs/adapter-node-http'
import FSPersister from '@pollyjs/persister-fs'
import fetch from 'node-fetch'

// https://github.com/mui-org/material-ui/issues/21293
export const withoutMuiID = (wrapper) => wrapper.html().replace(/id="mui-[0-9]*"/g, '').replace(/aria-labelledby="(mui-[0-9]* *)*"/g, '')

Enzyme.configure({ adapter: new Adapter() })

global.fetch = fetch

// See: https://github.com/vuejs/vue-test-utils/issues/974
global.requestAnimationFrame = cb => cb()
