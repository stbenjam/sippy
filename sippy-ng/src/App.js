import './App.css'
import { createTheme, makeStyles, useTheme } from '@material-ui/core/styles'
import { CssBaseline, Grid, MuiThemeProvider } from '@material-ui/core'
import { JobAnalysis } from './jobs/JobAnalysis'
import { QueryParamProvider } from 'use-query-params'
import { relativeTime } from './helpers'
import { Route, Switch } from 'react-router-dom'
import { TestAnalysis } from './tests/TestAnalysis'
import Alert from '@material-ui/lab/Alert'
import AppBar from '@material-ui/core/AppBar'
import BuildClusterDetails from './build_clusters/BuildClusterDetails'
import BuildClusterOverview from './build_clusters/BuildClusterOverview'
import ChevronLeftIcon from '@material-ui/icons/ChevronLeft'
import ChevronRightIcon from '@material-ui/icons/ChevronRight'
import clsx from 'clsx'
import Drawer from '@material-ui/core/Drawer'
import IconButton from '@material-ui/core/IconButton'
import Install from './releases/Install'
import Jobs from './jobs/Jobs'
import MenuIcon from '@material-ui/icons/Menu'
import PayloadStream from './releases/PayloadStream'
import PayloadStreams from './releases/PayloadStreams'
import React, { useEffect } from 'react'
import ReleaseOverview from './releases/ReleaseOverview'
import ReleasePayloadDetails from './releases/ReleasePayloadDetails'
import ReleasePayloads from './releases/ReleasePayloads'
import Sidebar from './components/Sidebar'
import Tests from './tests/Tests'
import Toolbar from '@material-ui/core/Toolbar'
import Typography from '@material-ui/core/Typography'
import Upgrades from './releases/Upgrades'
import VariantStatus from './jobs/VariantStatus'

const drawerWidth = 240

const useStyles = makeStyles((theme) => ({
  root: {
    display: 'flex',
    flexGrow: 1,
  },
  title: {
    flexGrow: 1,
  },
  appBar: {
    transition: theme.transitions.create(['margin', 'width'], {
      easing: theme.transitions.easing.sharp,
      duration: theme.transitions.duration.leavingScreen,
    }),
  },
  appBarShift: {
    width: `calc(100% - ${drawerWidth}px)`,
    marginLeft: drawerWidth,
    transition: theme.transitions.create(['margin', 'width'], {
      easing: theme.transitions.easing.easeOut,
      duration: theme.transitions.duration.enteringScreen,
    }),
  },
  backdrop: {
    zIndex: theme.zIndex.drawer + 1,
    color: '#fff',
  },
  menuButton: {
    marginRight: theme.spacing(2),
  },
  hide: {
    display: 'none',
  },
  drawer: {
    width: drawerWidth,
    flexShrink: 0,
  },
  drawerPaper: {
    width: drawerWidth,
  },
  drawerHeader: {
    display: 'flex',
    alignItems: 'center',
    padding: theme.spacing(0, 1),
    // necessary for content to be below app bar
    ...theme.mixins.toolbar,
    justifyContent: 'flex-end',
  },
  content: {
    maxWidth: '100%',
    flexGrow: 1,
    padding: theme.spacing(3),
    transition: theme.transitions.create('margin', {
      easing: theme.transitions.easing.sharp,
      duration: theme.transitions.duration.leavingScreen,
    }),
    marginLeft: -drawerWidth,
  },
  contentShift: {
    transition: theme.transitions.create('margin', {
      easing: theme.transitions.easing.easeOut,
      duration: theme.transitions.duration.enteringScreen,
    }),
    marginLeft: 0,
  },
}))

// Default theme:
const lightMode = {
  palette: {
    type: 'light',
  },
}

export const CapabilitiesContext = React.createContext([])

export default function App(props) {
  const classes = useStyles()
  const theme = useTheme()

  const [lastUpdated, setLastUpdated] = React.useState(null)
  const [drawerOpen, setDrawerOpen] = React.useState(true)
  const [isLoaded, setLoaded] = React.useState(false)
  const [releases, setReleases] = React.useState([])
  const [capabilities, setCapabilities] = React.useState([])
  const [fetchError, setFetchError] = React.useState('')

  const fetchData = () => {
    Promise.all([
      fetch(process.env.REACT_APP_API_URL + '/api/releases'),
      fetch(process.env.REACT_APP_API_URL + '/api/capabilities'),
    ])
      .then(([releases, capabilities]) => {
        if (releases.status !== 200) {
          throw new Error('server returned ' + releases.status)
        }

        if (capabilities.status !== 200) {
          throw new Error('server returned ' + capabilities.status)
        }
        return Promise.all([releases.json(), capabilities.json()])
      })
      .then(([releases, capabilities]) => {
        setReleases(releases['releases'])
        setCapabilities(capabilities)
        setLastUpdated(new Date(releases.last_updated))
        setLoaded(true)
      })
      .catch((error) => {
        setLoaded(true)
        setFetchError('could not retrieve data:' + error)
      })
  }

  useEffect(() => {
    if (!isLoaded) {
      fetchData()
    }
  })

  const handleDrawerOpen = () => {
    setDrawerOpen(true)
  }

  const handleDrawerClose = () => {
    setDrawerOpen(false)
  }

  if (!isLoaded) {
    return <Typography>Loading...</Typography>
  }

  let landingPage = ''
  if (fetchError !== '') {
    landingPage = <Alert severity="error">{fetchError}</Alert>
  } else if (releases.length > 0) {
    landingPage = <ReleaseOverview key={releases[0]} release={releases[0]} />
  } else {
    landingPage = 'No data.'
  }

  return (
    <CapabilitiesContext.Provider value={capabilities}>
      <MuiThemeProvider theme={createTheme(lightMode)}>
        <CssBaseline />
        <QueryParamProvider ReactRouterRoute={Route}>
          <div className={classes.root}>
            <AppBar
              position="fixed"
              className={clsx(classes.appBar, {
                [classes.appBarShift]: drawerOpen,
              })}
            >
              <Toolbar edge="start">
                <IconButton
                  color="inherit"
                  aria-label="open drawer"
                  onClick={handleDrawerOpen}
                  edge="start"
                  className={clsx(
                    classes.menuButton,
                    drawerOpen && classes.hide
                  )}
                >
                  <MenuIcon />
                </IconButton>
                <Grid
                  container
                  justifyContent="space-between"
                  alignItems="center"
                >
                  <Typography variant="h6" className={classes.title}>
                    Sippy
                  </Typography>
                  Last updated{' '}
                  {lastUpdated !== null ? relativeTime(lastUpdated) : 'unknown'}
                </Grid>
              </Toolbar>
            </AppBar>

            <Drawer
              className={classes.drawer}
              variant="persistent"
              anchor="left"
              open={drawerOpen}
              classes={{
                paper: classes.drawerPaper,
              }}
            >
              <div className={classes.drawerHeader}>
                <IconButton onClick={handleDrawerClose}>
                  {theme.direction === 'ltr' ? (
                    <ChevronLeftIcon />
                  ) : (
                    <ChevronRightIcon />
                  )}
                </IconButton>
              </div>
              <Sidebar releases={releases} />
            </Drawer>

            <main
              className={clsx(classes.content, {
                [classes.contentShift]: drawerOpen,
              })}
            >
              <div className={classes.drawerHeader} />

              {/* eslint-disable react/prop-types */}
              <Switch>
                <Route path="/about">
                  <p>Hello, world!</p>
                </Route>

                <Route
                  path="/release/:release/tags/:tag"
                  render={(props) => (
                    <ReleasePayloadDetails
                      key={'release-details-' + props.match.params.release}
                      release={props.match.params.release}
                      releaseTag={props.match.params.tag}
                    />
                  )}
                />

                <Route
                  path="/release/:release/streams/:arch/:stream"
                  render={(props) => (
                    <PayloadStream
                      release={props.match.params.release}
                      arch={props.match.params.arch}
                      stream={props.match.params.stream}
                    />
                  )}
                />

                <Route
                  path="/release/:release/streams"
                  render={(props) => (
                    <PayloadStreams
                      key={'release-streams-' + props.match.params.release}
                      release={props.match.params.release}
                    />
                  )}
                />

                <Route
                  path="/release/:release/tags"
                  render={(props) => (
                    <ReleasePayloads
                      key={'release-tags-' + props.match.params.release}
                      release={props.match.params.release}
                    />
                  )}
                />

                <Route
                  path="/release/:release"
                  render={(props) => (
                    <ReleaseOverview
                      key={'release-overview-' + props.match.params.release}
                      release={props.match.params.release}
                    />
                  )}
                />

                <Route
                  path="/variants/:release/:variant"
                  render={(props) => (
                    <VariantStatus
                      release={props.match.params.release}
                      variant={props.match.params.variant}
                    />
                  )}
                />

                <Route
                  path="/jobs/:release/analysis"
                  render={(props) => (
                    <JobAnalysis release={props.match.params.release} />
                  )}
                />

                <Route
                  path="/jobs/:release"
                  render={(props) => (
                    <Jobs
                      key={'jobs-' + props.match.params.release}
                      title={'Job results for ' + props.match.params.release}
                      release={props.match.params.release}
                    />
                  )}
                />

                <Route
                  path="/tests/:release/analysis"
                  render={(props) => (
                    <TestAnalysis release={props.match.params.release} />
                  )}
                />

                <Route
                  path="/tests/:release"
                  render={(props) => (
                    <Tests
                      key={'tests-' + props.match.params.release}
                      release={props.match.params.release}
                    />
                  )}
                />

                <Route
                  path="/upgrade/:release"
                  render={(props) => (
                    <Upgrades
                      key={'upgrades-' + props.match.params.release}
                      release={props.match.params.release}
                    />
                  )}
                />

                <Route
                  path="/install/:release"
                  render={(props) => (
                    <Install
                      key={'install-' + props.match.params.release}
                      release={props.match.params.release}
                    />
                  )}
                />

                <Route
                  path="/build_clusters/:cluster"
                  render={(props) => (
                    <BuildClusterDetails
                      key={'cluster-' + props.match.params.cluster}
                      cluster={props.match.params.cluster}
                    />
                  )}
                />

                <Route
                  path="/build_clusters"
                  render={() => <BuildClusterOverview />}
                />

                <Route path="/">{landingPage}</Route>
              </Switch>
              {/* eslint-enable react/prop-types */}
            </main>
          </div>
        </QueryParamProvider>
      </MuiThemeProvider>
    </CapabilitiesContext.Provider>
  )
}
