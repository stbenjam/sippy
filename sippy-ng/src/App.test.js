/** @jest-environment setup-polly-jest/jest-environment-node */

import 'jsdom-global/register'
import App from './App'
import React from 'react'
import { act } from '@testing-library/react'
import { mount } from 'enzyme'
import { setupPolly } from 'setup-polly-jest'
import toJson from 'enzyme-to-json'
import { withoutMuiID } from './setupTests'
import path from 'path'

jest.useRealTimers()

describe('app', () => {
  const context = setupPolly({})

  beforeEach(() => {
    context.polly.configure({
      adapters: [require('@pollyjs/adapter-node-http')],
      persister: require('@pollyjs/persister-fs'),
      persisterOptions: {
        fs: {
          recordingsDir: path.resolve(__dirname, '../__recordings__')
        }
      }
    }
    )
  })

  it('should fetch correctly', async () => {
    let wrapper
    await act(async () => {
      wrapper = mount(<App/>)
    })

    expect(wrapper.exists).toBeTruthy()
    expect(wrapper.find('p').text()).toContain('Loading...')

    // Wait for data to load...
    await act(async () => {
      await new Promise(resolve => setTimeout(resolve, 0))
    })
    wrapper.update()

    expect(withoutMuiID(wrapper)).toMatchSnapshot()
  })
})
