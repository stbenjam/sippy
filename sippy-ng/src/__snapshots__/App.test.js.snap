// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`app should fetch correctly 1`] = `"<div class=\\"makeStyles-root-1\\"><header class=\\"MuiPaper-root MuiAppBar-root MuiAppBar-positionFixed MuiAppBar-colorPrimary makeStyles-appBar-3 mui-fixed MuiPaper-elevation4\\"><div class=\\"MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters\\" edge=\\"start\\"><button class=\\"MuiButtonBase-root MuiIconButton-root makeStyles-menuButton-6 MuiIconButton-colorInherit MuiIconButton-edgeStart\\" tabindex=\\"0\\" type=\\"button\\" aria-label=\\"open drawer\\"><span class=\\"MuiIconButton-label\\"><svg class=\\"MuiSvgIcon-root\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\"><path d=\\"M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z\\"></path></svg></span><span class=\\"MuiTouchRipple-root\\"></span></button><div class=\\"MuiGrid-root MuiGrid-container MuiGrid-align-items-xs-center MuiGrid-justify-content-xs-space-between\\"><h6 class=\\"MuiTypography-root makeStyles-title-2 MuiTypography-h6\\">Sippy</h6>Last updated 12 hours ago</div></div></header><div class=\\"MuiDrawer-root MuiDrawer-docked makeStyles-drawer-8\\"><div class=\\"MuiPaper-root MuiDrawer-paper makeStyles-drawerPaper-9 MuiDrawer-paperAnchorLeft MuiDrawer-paperAnchorDockedLeft MuiPaper-elevation0\\" style=\\"visibility: hidden; webkit-transform: translateX(-0px); transform: translateX(-0px);\\"><div class=\\"makeStyles-drawerHeader-10\\"><button class=\\"MuiButtonBase-root MuiIconButton-root\\" tabindex=\\"0\\" type=\\"button\\"><span class=\\"MuiIconButton-label\\"><svg class=\\"MuiSvgIcon-root\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\"><path d=\\"M15.41 7.41L14 6l-6 6 6 6 1.41-1.41L10.83 12z\\"></path></svg></span><span class=\\"MuiTouchRipple-root\\"></span></button></div><ul class=\\"MuiList-root MuiList-padding\\"><a class=\\"MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button\\" tabindex=\\"0\\" role=\\"button\\" aria-disabled=\\"false\\" href=\\"/\\"><div class=\\"MuiListItemIcon-root\\"><svg class=\\"MuiSvgIcon-root\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\"><path d=\\"M10 20v-6h4v6h5v-8h3L12 3 2 12h3v8z\\"></path></svg></div><div class=\\"MuiListItemText-root\\"><span class=\\"MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock\\">Home</span></div><span class=\\"MuiTouchRipple-root\\"></span></a></ul><hr class=\\"MuiDivider-root\\"><ul class=\\"MuiList-root MuiList-padding MuiList-subheader\\"><div class=\\"MuiListSubheader-root MuiListSubheader-sticky MuiListSubheader-gutters\\" id=\\"releases\\">Releases</div><div class=\\"MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button\\" tabindex=\\"0\\" role=\\"button\\" aria-disabled=\\"false\\"><svg class=\\"MuiSvgIcon-root\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\"><path d=\\"M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z\\"></path></svg><div class=\\"MuiListItemText-root\\"><span class=\\"MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock\\">4.9</span></div><span class=\\"MuiTouchRipple-root\\"></span></div></ul><hr class=\\"MuiDivider-root\\"><ul class=\\"MuiList-root MuiList-padding MuiList-subheader\\"><div class=\\"MuiListSubheader-root MuiListSubheader-sticky MuiListSubheader-gutters\\" id=\\"resources\\">Resources</div><a class=\\"MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button\\" tabindex=\\"0\\" aria-disabled=\\"false\\" href=\\"https://testgrid.k8s.io/redhat\\" target=\\"_blank\\"><div class=\\"MuiListItemIcon-root\\"><svg class=\\"MuiSvgIcon-root\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\"><path d=\\"M19 3H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2zM9 17H7v-7h2v7zm4 0h-2V7h2v10zm4 0h-2v-4h2v4z\\"></path></svg></div><div class=\\"MuiListItemText-root\\"><span class=\\"MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock\\">TestGrid</span></div><span class=\\"MuiTouchRipple-root\\"></span></a><a class=\\"MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button\\" tabindex=\\"0\\" aria-disabled=\\"false\\" href=\\"https://amd64.ocp.releases.ci.openshift.org/\\" target=\\"_blank\\"><div class=\\"MuiListItemIcon-root\\"><svg class=\\"MuiSvgIcon-root\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\"><path d=\\"M23 12l-2.44-2.78.34-3.68-3.61-.82-1.89-3.18L12 3 8.6 1.54 6.71 4.72l-3.61.81.34 3.68L1 12l2.44 2.78-.34 3.69 3.61.82 1.89 3.18L12 21l3.4 1.46 1.89-3.18 3.61-.82-.34-3.68L23 12zm-10 5h-2v-2h2v2zm0-4h-2V7h2v6z\\"></path></svg></div><div class=\\"MuiListItemText-root\\"><span class=\\"MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock\\">Release Controller</span></div><span class=\\"MuiTouchRipple-root\\"></span></a><div class=\\"MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button\\" tabindex=\\"0\\" role=\\"button\\" aria-disabled=\\"false\\"><div class=\\"MuiListItemIcon-root\\"><svg class=\\"MuiSvgIcon-root\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\"><path d=\\"M20 8h-2.81c-.45-.78-1.07-1.45-1.82-1.96L17 4.41 15.59 3l-2.17 2.17C12.96 5.06 12.49 5 12 5c-.49 0-.96.06-1.41.17L8.41 3 7 4.41l1.62 1.63C7.88 6.55 7.26 7.22 6.81 8H4v2h2.09c-.05.33-.09.66-.09 1v1H4v2h2v1c0 .34.04.67.09 1H4v2h2.81c1.04 1.79 2.97 3 5.19 3s4.15-1.21 5.19-3H20v-2h-2.09c.05-.33.09-.66.09-1v-1h2v-2h-2v-1c0-.34-.04-.67-.09-1H20V8zm-6 8h-4v-2h4v2zm0-4h-4v-2h4v2z\\"></path></svg></div><div class=\\"MuiListItemText-root\\"><span class=\\"MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock\\">Search Bugzilla</span></div><span class=\\"MuiTouchRipple-root\\"></span></div><a class=\\"MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button\\" tabindex=\\"0\\" aria-disabled=\\"false\\" target=\\"_blank\\" href=\\"https://www.github.com/openshift/sippy\\"><div class=\\"MuiListItemIcon-root\\"><svg class=\\"MuiSvgIcon-root\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\"><path d=\\"M12 .3a12 12 0 0 0-3.8 23.4c.6.1.8-.3.8-.6v-2c-3.3.7-4-1.6-4-1.6-.6-1.4-1.4-1.8-1.4-1.8-1-.7.1-.7.1-.7 1.2 0 1.9 1.2 1.9 1.2 1 1.8 2.8 1.3 3.5 1 0-.8.4-1.3.7-1.6-2.7-.3-5.5-1.3-5.5-6 0-1.2.5-2.3 1.3-3.1-.2-.4-.6-1.6 0-3.2 0 0 1-.3 3.4 1.2a11.5 11.5 0 0 1 6 0c2.3-1.5 3.3-1.2 3.3-1.2.6 1.6.2 2.8 0 3.2.9.8 1.3 1.9 1.3 3.2 0 4.6-2.8 5.6-5.5 5.9.5.4.9 1 .9 2.2v3.3c0 .3.1.7.8.6A12 12 0 0 0 12 .3\\"></path></svg></div><div class=\\"MuiListItemText-root\\"><span class=\\"MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock\\">GitHub Repo</span></div><span class=\\"MuiTouchRipple-root\\"></span></a><hr class=\\"MuiDivider-root\\"><div align=\\"center\\"><div align=\\"center\\"><p class=\\"MuiTypography-root MuiTypography-body1\\" aria-haspopup=\\"true\\"><img class=\\"Sippy-logo\\" src=\\"sippy.svg\\" alt=\\"CIPI (Continuous Integration Private Investigator) aka Sippy.\\"><br></p></div></div></ul></div></div><main class=\\"makeStyles-content-11\\"><div class=\\"makeStyles-drawerHeader-10\\"></div></main></div>"`;
